### DISCUSSION TITLE
Umami collection being blocked, resulting in less users and impressions being shown

### DISCUSSION THREAD

**Discussion by Fawesum** [umami-software/umami#1749] [General]
So I installed Umami because I wanted to get traffic numbers for those who block Google Analytics with adblockers. The first year or so this went well and we saw that around 30% of our users blocked GA – we finally had correct numbers!

However, the last few months I've seen Umami report way less users and impressions than Google Analytics. Like 30-40% more than Umami.

It seems Umami is now being blocked by the likes of uBlock. The default umami.js file is blocked, but even if I rename it to something not on the blocked list, the collection of data is blocked:

`api/collect net::ERR_BLOCKED_BY_CLIENT`

It's the XMLHttpRequest that is being blocked:

`function(t, e, n) {
                        var a = new XMLHttpRequest;
                        a.open("POST", t, !0),
                        a.setRequestHeader("Content-Type", "application/json"),
                        a.onreadystatechange = function() {
                            4 === a.readyState && n && n(a.response)
                        }
                        ,
                        a.send(JSON.stringify(e))
                    }(N + "/api/collect", {
                        type: t,
                        payload: i
                    }, (function(t) {
                        return k && f.setItem(a, t)
                    }
                    ))`

Any ideas on how to counteract this?

**Comment by sdvcrx** [comment]
You can try setting the environment variable `COLLECT_API_ENDPOINT` and rebuild umami (default value is `/api/collect`)

**Comment by hkamran80** [comment]
I use a Cloudflare Worker, as described in https://github.com/umami-software/umami/discussions/1026#discussioncomment-2581700.
